test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/ListPackageTests.cs

```diff
@@ -290,6 +291,178 @@ static async Task RestoreProjectAsync(ISettings settings,
+            auditSource.AllowInsecureConnections = true;
+
+            using (SimpleTestPathContext pathContext = new SimpleTestPathContext())
+            {
+                string projectFolder = Path.Combine(pathContext.SolutionRoot, "MyProject");
+                string csprojPath = Path.Combine(projectFolder, "MyProject.csproj");
+                string objFolder = Path.Combine(projectFolder, "obj");
+                string assetsPath = Path.Combine(objFolder, "project.assets.json");
+
+                //setup package
+                var packageContext = new SimpleTestPackageContext()
+                {
+                    Id = "mypkg",
+                    Version = "5.8.2",
+                    Nuspec = XDocument.Parse($@"<?xml version=""1.0"" encoding=""utf-8""?>
+                        <package>
+                        <metadata>
+                            <id>mypkg</id>
+                            <version>5.8.2</version>
+                            <title />
+                            <frameworkAssemblies>
+                                <frameworkAssembly assemblyName=""System.Runtime"" />
+                            </frameworkAssemblies>
+                            <contentFiles>
+                                <files include=""lib/net45/mypkg.dll"" copyToOutput=""true"" flatten=""false"" />
+                            </contentFiles>
+                        </metadata>
+                        </package>")
+                };
+                await SimpleTestPackageUtility.CreatePackagesAsync(pathContext.PackageSource, packageContext);
+
+                // Define the content for csproj
+                var csprojContent = @"
+<Project Sdk=""Microsoft.NET.Sdk"">
+  <PropertyGroup>
+    <OutputType>Library</OutputType>
+    <TargetFramework>net5.0</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include=""mypkg"" Version=""5.8.2"" />
+  </ItemGroup>
+</Project>";
+
+                // Define the content for assets.json
+                var assetsContent = @"
+{
+  ""version"": 3,
+  ""targets"": {
+    "".NETCoreApp,Version=v5.0"": {
+      ""mypkg/5.8.2"": {
+        ""type"": ""package"",
+        ""compile"": {
+          ""lib/net5.0/mypkg.dll"": {}
+        },
+        ""runtime"": {
+          ""lib/net5.0/mypkg.dll"": {}
+        }
+      }
+    }
+  },
+  ""libraries"": {
+    ""mypkg/5.8.2"": {
+      ""sha512"": ""<hash-value>"",
+      ""type"": ""package"",
+      ""path"": ""mypkg/5.8.2""
+    }
+  },
+  ""project"": {
+    ""version"": ""5.0"",
+    ""restore"": {
+      ""projectStyle"": ""PackageReference""
+    },
+    ""frameworks"": {
+      ""net5.0"": {
+        ""dependencies"": {
+          ""mypkg"": {
+            ""version"": ""5.8.2"",
+            ""type"": ""direct""
+          }
+        }
+      }
+    }
+  }
+}";
+                if (!Directory.Exists(projectFolder))
+                {
+                    Directory.CreateDirectory(projectFolder);
```

Comment:
`Directory.CreateDirectory` does not throw if the path already exists, and considering `SimpleTestPathContext` creates a random directory for the test to run in, we know that this directory does not exist.

This test is very long, making it hard to review for correctness, so little things like removing unnecessary if blocks can help shorten it a bit.